src/link/batch/batching.ts

```diff
@@ -58,48 +65,70 @@ export class OperationBatcher {
   }
 
   public enqueueRequest(request: BatchableRequest): Observable<FetchResult> {
-    const requestCopy = {
+    const requestCopy: QueuedRequest = {
       ...request,
+      next: [],
+      error: [],
+      complete: [],
+      subscribers: new Set,
     };
```

Comment:
You restored `BatchableRequest` as it was before, and I don't mind, but I did make the original change to it because I could not figure what those optional properties could be used for. I don't see what use-case could require to queue a request while already providing some pre-subscribed subscribers... :woman_shrugging: 

I am 100% sure this possibility is never used in this package at all. But maybe there are usages out there in the wild that do use this ? :thinking:

As a matter of fact, I suspect that passing pre-subscribed subscribers this way would probably mess up indices were those callbacks would be called leading to a `server returned results with length ...` error :thinking: 

Anyway, if you really want to keep `BatchableRequest` as it was before, then you probably want to preserve those optional properties when creating the copy:

```suggestion
    const requestCopy: QueuedRequest = {
      next: [],
      error: [],
      complete: [],
      subscribers: new Set,
      ...request,
    };
```

Reply:
I did that in https://github.com/apollographql/apollo-client/pull/9248/commits/29fb39ed033614aa2d441034acda32c51545171b